### PR TITLE
Podman stats error

### PR DESCRIPTION
**Pull request by rogercoll**
Add correct data structure to prevent Unmarshal error from containerStats structure.

Solves #9397

### CODE REVIEW

**Comment by rogercoll** [comment]
Requirement for Podman refactor PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9419

File: receiver/podmanreceiver/podman_client.go [hunk 1]
```diff
@@ -30,6 +30,10 @@ import (
 	"go.uber.org/zap"
 )
 
+var (
+	ErrNoStatsFound = fmt.Errorf("No stats found")
```
Comment (mx-psi) [inline receiver/podmanreceiver/podman_client.go:34]: Does this need to be public?

**Review by mx-psi — APPROVED** [review]

File: receiver/podmanreceiver/podman_client.go (hunk 1 above)
Comment (rogercoll) [reply receiver/podmanreceiver/podman_client.go]: No need, fixed. 

Thank you

**Review by jpkrohling — COMMENTED** [review]
Looking at the PR alone, it's hard to tell what the problem it is solving and whether it is actually solving the problem. Could you please add a test that would fail on the current code base and work when the PR is applied?

**Comment by rogercoll** [comment]
Hi @jpkrohling ,

The PR tries to solve the following error message:
`"json: cannot unmarshal object into Go struct field containerStatsReport.Error of type string"`

For example, if you query for a non existing container you get the following output:
```
# curl --unix-socket /run/user/1000/podman/podman.sock http://d/v3.4.0/libpod/containers/stats?containers=notacontainer
{"Error":{},"Stats":null}
```

Instead, if we query for a valid container:
```
# curl --unix-socket /run/user/1000/podman/podman.sock http://d/v3.4.0/libpod/containers/stats?containers=e6af5805edae
{"Error":null,"Stats":[{"AvgCPU":42.04781177856639,"ContainerID":"e6af5805edae6c950003abd5451808b277b67077e400f0a6f69d01af116ef014","Name":"charming_sutherland","PerCPU":null,"CPU":42.04781177856639,"CPUNano":309165846000,"CPUSystemNano":54515674,"SystemNano":1650912926385978706,"MemUsage":27717632,"MemLimit":7942234112,"MemPerc":0.34899036730888044,"NetInput":430,"NetOutput":330,"BlockInput":0,"BlockOutput":0,"PIDs":118,"UpTime":309165846000,"Duration":309165846000}]}
```

What I understood from the [documentation](https://docs.podman.io/en/latest/_static/api.html#operation/ContainersStatsAllLibpod) is that if there is an error on the API, podman returns an object, not a string. 

The current code base does not make queries filtering containers and that error might be a bit difficult to reproduce (e.g internal API error), nonetheless I will definitely take a look at the code and apply some tests.

**Review by jpkrohling — APPROVED** [review]

File: receiver/podmanreceiver/podman_client_test.go [hunk 2]
```diff
@@ -75,3 +78,95 @@ func TestWatchingTimeouts(t *testing.T) {
...
+		UpTime:        309165846000 * time.Nanosecond,
+		Duration:      309165846000,
+	}
+
+	stats, err := cli.stats(context.Background())
+	assert.NoError(t, err)
+	assert.Equal(t, expectedStats, stats[0])
+}
+
+func TestStatsError(t *testing.T) {
```
Comment (TylerHelmuth) [inline receiver/podmanreceiver/podman_client_test.go:138]: Nice test

**Review by TylerHelmuth — APPROVED** [review]
@open-telemetry/collector-contrib-approvers please review

**Comment by dmitryax** [comment]
@rogercoll please fix liner issues

**Comment by rogercoll** [comment]
@dmitryax Linter issues fixed. Thanks

**Review by pmm-sumo — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**